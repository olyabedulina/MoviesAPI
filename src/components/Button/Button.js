import React from 'react'
import PropTypes from 'prop-types'

import CM from './styles.pcss'

const Button = ({
    className = '',
    kind = 'main',
    children,
    onClick = Function.prototype
}) => {
    function handleClick() {
        onClick();
    }

    return <button
        className={`${className} ${CM.button} ${CM["kind-" + kind] || ''}`}
        type="button"
        onClick={handleClick}>
            {children}
        </button>
}

Button.propTypes = {
    className: PropTypes.string,
    kind: PropTypes.string,
    children: PropTypes.node,
    onClick: PropTypes.func
};

export default Button