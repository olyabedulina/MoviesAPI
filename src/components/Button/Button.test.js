import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('Button component test:', () => {
    it('Testing "+ Add Movie" Button component', () => {
        const actual = renderer
            .create(<Button kind="aux" className="buttonAddMovie">+ Add Movie</Button>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Testing "Search" Button component', () => {
        const actual = renderer
            .create(<Button type="submit" className="formButton">Search</Button>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Testing "Reset" Button component', () => {
        const actual = renderer
            .create(<Button kind='alt' className="modalFooterButton">Reset</Button>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Testing "Submit" Button component', () => {
        const actual = renderer
            .create(<Button type="submit" kind='main' className="modalFooterButton">Submit</Button>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Testing "Go Back To Home" Button component', () => {
        const actual = renderer
            .create(<Button kind='alt' className="pageNotFoundButton">Go Back To Home</Button>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });
});
