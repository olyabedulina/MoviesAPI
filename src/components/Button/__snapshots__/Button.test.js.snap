// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test: Testing "+ Add Movie" Button component 1`] = `
<button
  className="buttonAddMovie button kind-aux"
  onClick={[Function]}
  type="button"
>
  + Add Movie
</button>
`;

exports[`Button component test: Testing "Go Back To Home" Button component 1`] = `
<button
  className="pageNotFoundButton button kind-alt"
  onClick={[Function]}
  type="button"
>
  Go Back To Home
</button>
`;

exports[`Button component test: Testing "Reset" Button component 1`] = `
<button
  className="modalFooterButton button kind-alt"
  onClick={[Function]}
  type="button"
>
  Reset
</button>
`;

exports[`Button component test: Testing "Search" Button component 1`] = `
<button
  className="formButton button kind-main"
  onClick={[Function]}
  type="submit"
>
  Search
</button>
`;

exports[`Button component test: Testing "Submit" Button component 1`] = `
<button
  className="modalFooterButton button kind-main"
  onClick={[Function]}
  type="submit"
>
  Submit
</button>
`;
